src/main/java/se/sundsvall/supportmanagement/service/CommunicationService.java

```diff
@@ -1,67 +1,105 @@
+
 	private static final boolean ASYNCHRONOUSLY = true;
 
-	@Autowired
-	private ErrandsRepository repository;
+	private final ErrandsRepository errandsRepository;
+
+	private final MessagingClient messagingClient;
+
+	private final CommunicationRepository communicationRepository;
 
-	@Autowired
-	private MessagingClient messagingClient;
+	private final CommunicationAttachmentRepository communicationAttachmentRepository;
+
+	public CommunicationService(final ErrandsRepository errandsRepository,
+		final MessagingClient messagingClient, final CommunicationRepository communicationRepository,
+		final CommunicationAttachmentRepository communicationAttachmentRepository) {
+		this.errandsRepository = errandsRepository;
+		this.messagingClient = messagingClient;
+		this.communicationRepository = communicationRepository;
+		this.communicationAttachmentRepository = communicationAttachmentRepository;
+	}
 
 
 	public List<Communication> readCommunications(final String namespace, final String municipalityId, final String id) {
-		fetchEntity(id, namespace, municipalityId);
+		final var entity = fetchEntity(id, namespace, municipalityId);
 
-		return Collections.emptyList();
+		return CommunicationMapper.toCommunications(communicationRepository.findByErrandNumber(entity.getErrandNumber()));
 	}
 
-	public void updateViewedStatus(final String namespace, final String municipalityId, final String id, final String messageID, final boolean isViewed) {
+	public void updateViewedStatus(final String namespace, final String municipalityId, final String id, final String communicationId, final boolean isViewed) {
 		fetchEntity(id, namespace, municipalityId);
+
+		final var message = communicationRepository
+			.findById(communicationId)
+			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, "Communication with id %s not found".formatted(communicationId)));
+
+		message.setViewed(isViewed);
+		communicationRepository.save(message);
 	}
 
 	public void getMessageAttachmentStreamed(final String attachmentID, final HttpServletResponse response) {
 
-		// TODO: Implement this method
+		try {
+			final var attachmentEntity = communicationAttachmentRepository
+				.findById(attachmentID)
+				.orElseThrow(() -> Problem.valueOf(Status.NOT_FOUND, "MessageAttachment not found"));
+
+			final var file = attachmentEntity.getAttachmentData().getFile();
 
+			response.addHeader(CONTENT_TYPE, attachmentEntity.getContentType());
+			response.addHeader(CONTENT_DISPOSITION, "attachment; filename=\"" + attachmentEntity.getName() + "\"");
+			response.setContentLength((int) file.length());
+			StreamUtils.copy(file.getBinaryStream(), response.getOutputStream());
+		} catch (final IOException | SQLException e) {
+			throw Problem.valueOf(Status.INTERNAL_SERVER_ERROR, "%s occurred when copying file with attachment id '%s' to response: %s".formatted(e.getClass().getSimpleName(), attachmentID, e.getMessage()));

```

Comment:
```suggestion
			throw Problem.valueOf(INTERNAL_SERVER_ERROR, "%s occurred when copying file with attachment id '%s' to response: %s".formatted(e.getClass().getSimpleName(), attachmentID, e.getMessage()));

```

Inconsistent use of static imports